### PR TITLE
[CDAP-17246] Fixes js error while importing from hub or adding pipeli…

### PR DESCRIPTION
**Pull request by ajainarayanan**
…ne from wrangler

- Fixes plugins with spaces in labels to show preview data
- Fixes selecting nodes/connections using keyboard to copy/paste
- Fixes e2e tests failing in 6.2.2
- Fixes nodes copied from 6.1.x version of CDAP to pasteable in 6.2.x

Cherry-pick of https://github.com/cdapio/cdap/pull/12768

### CODE REVIEW

**Comment by njbriggs** [comment]
There are several differences between this PR and the one it's cherry-picked from. Is that expected?

**Comment by ajainarayanan** [comment]
> There are several differences between this PR and the one it's cherry-picked from. Is that expected?

@njbriggs Yes that is correct. The major one being tests that are not included in this PR. Those tests have been individually added to the develop branch that I didn't need to add again.

File: cdap-ui/app/directives/dag-plus/services/stores/nodes-store.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ class DAGPlusPlusNodesStore {
       nodeConfig.name = nodeConfig.plugin.label + '-' + this.uuid.v4();
     }
     if (!nodeConfig.id) {
-      nodeConfig.id = nodeConfig.plugin.label.replace(/[ \/]/g, '-');
+      nodeConfig.id = nodeConfig.plugin.label.replace(/[ \/]/g, '-') + '-' + this.uuid.v4();
```
Comment (njbriggs) [inline cdap-ui/app/directives/dag-plus/services/stores/nodes-store.js:133]: Does this still need the hyphen-replacement fix?

**Review by njbriggs — COMMENTED** [review]
Overall, looks good. Just wondering if this needs the fix for Avro schema validation, and if so, if it should be added to this PR or go in later.

File: cdap-ui/app/directives/dag-plus/services/stores/nodes-store.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/directives/dag-plus/services/stores/nodes-store.js:133]: This is done only on id. This shouldn't affect the pipeline. This is needed only for multiselect.

**Comment by ajainarayanan** [comment]
> Overall, looks good. Just wondering if this needs the fix for Avro schema validation, and if so, if it should be added to this PR or go in later.

@njbriggs I am not sure. Do you see any failures related to schema validation.

**Comment by njbriggs** [comment]
> Do you see any failures related to schema validation.

@ajainarayanan No, I was just wondering if the code I commented on above needed the regex to remove the hyphens.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**